Feature/tests

the-man-w-laughs opened the pull request:
This pull request contains unit tests for Identity microservice and for TeamHub microservice.
Unit tests are based on XUnit.

the-man-w-laughs committed:
Add Identity.Tests

the-man-w-laughs committed:
Merge branch 'main' into feature/tests

the-man-w-laughs committed:
Add unit tests for Identity services.

the-man-w-laughs committed:
Add invokation count verification.

the-man-w-laughs committed:
Use Bogus generators.

the-man-w-laughs committed:
Add more unit tests

Major changes:
- add base code for UsersController tests;
- add tests for EmailConfirmationService.

the-man-w-laughs committed:
Add more tests

Major changes:
- complete UserController tests;
- add GetUserByEmail tests to UserServiceTests;
- refactor UserServiceTests;
- refactor UserController.

the-man-w-laughs committed:
Complete unit tests for Identity microservice

Major changes:
- rafactor UserControllerTests;
- add AppUserRepositoryHelper;
- add MapperHelper;
- add UserManagerHelper;
- add UserServiceHelper;
- add UserRepositoryTests;
- refactor EmailConfirmationServiceTests.

the-man-w-laughs committed:
Rename EmailServiceTests to EmailConfirmationServiceTest

the-man-w-laughs committed:
Add first test to TeamHub.Tests

the-man-w-laughs committed:
Add tests for CreateCommentCommandHandler.

the-man-w-laughs committed:
Add tests for DeleteCommentCommandHandler.

the-man-w-laughs committed:
Add tests for UpdateCommentCommandHandler.

the-man-w-laughs committed:
Add tests for GetAllTasksCommentsQueryHandler.

the-man-w-laughs committed:
Move fakers to separate classes.

katyalys commented on src/services/Identity/Identity.Tests/ControllerTests/UserControllerTests.cs:
```diff
@@ -0,0 +1,198 @@
+
+            _appUserDto = new Faker<AppUserDto>()
+                .RuleFor(u => u.Id, f => f.Random.Number(max: int.MaxValue))
+                .RuleFor(u => u.Email, f => f.Internet.Email())
```
make more explicit naming instead of u, f

katyalys commented on src/services/Identity/Identity.Tests/Helpers/UserManagerHelper.cs:
```diff
@@ -0,0 +1,107 @@
+                    (AppUser user, string password) =>
+                    {
+                        users.Add(user);
+                        return IdentityResult.Success;
```
empty line before return

katyalys commented on src/services/Identity/Identity.Tests/RepositoriesTests/UserRepositoryTests.cs:
```diff
@@ -0,0 +1,236 @@
+                store.Object,
+                null,
+                null,
+                null,
```
why do you pass null? why cant you mock all dependencies here?

katyalys commented on src/services/Identity/Identity.Tests/RepositoriesTests/UserRepositoryTests.cs:
```diff
@@ -0,0 +1,236 @@
+        public async void IsInRoleAsync_UserHasTargetRole_ReturnTrue()
+        {
+            // Arrange
+            var targetRole = "admin";
```
create enum or constant class

katyalys commented on src/services/Identity/Identity.Tests/ControllerTests/UserControllerTests.cs:
```diff
@@ -0,0 +1,198 @@
+            _userServiceHelper = new UserServiceHelper(_userService);
+
+            _appUserRegisterDto = new Faker<AppUserRegisterDto>()
+                .RuleFor(u => u.Email, f => f.Internet.Email())
```
move fakers to helper classes

the-man-w-laughs replied:
I made separate classes for fakers.

katyalys commented on src/services/Identity/Identity.Tests/ServiceTests/EmailConfirmationServiceTests.cs:
```diff
@@ -0,0 +1,135 @@
+            Mock.Of<IUserStore<AppUser>>(),
+            null,
+            null,
+            null,
```
same

katyalys commented on src/services/Identity/Identity.Tests/ServiceTests/EmailConfirmationServiceTests.cs:
```diff
@@ -0,0 +1,135 @@
+    {
+        // Arrange
+        var email = _faker.Internet.Email();
+        var token = "valid_token";
```
https://andrewlock.net/creating-parameterised-tests-in-xunit-with-inlinedata-classdata-and-memberdata/ read about inline data, and pass some of your parameters through it in some of your tests

katyalys commented on src/services/Identity/Identity.Tests/ServiceTests/UserServiceTests.cs:
```diff
@@ -0,0 +1,312 @@
+        _appUserRepositoryHelper = new AppUserRepositoryHelper(_appUserRepositoryMock);
+
+        _appUserFaker = new Faker<AppUser>()
+            .RuleFor(u => u.Id, f => f.Random.Number(max: int.MaxValue))
```
move to helper

the-man-w-laughs committed:
Fix review.

Major changes:
- add explicit lambda params naming;
- add empty line before returns;
- mock all dependencies for UserManager;
- start using inline data;
- add class to stode role names for tests.
